### PR TITLE
=htc #19058 prevent errors to be logged for http timeouts

### LINKED ISSUE #19058: Make Idle Timeout log DEBUG instead of exception in HTTP
As discused in https://github.com/akka/akka/issues/18982#issuecomment-160931684

### PR DESCRIPTION
**Pull request by jrudolph**
Fixes #19058, follow up to #19130.

/cc @danielwegener

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala [hunk 1]
```diff
@@ -85,7 +85,7 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
     connections.map {
       case Tcp.IncomingConnection(localAddress, remoteAddress, flow) ⇒
         val layer = serverLayer(settings, Some(remoteAddress), log)
-        val flowWithTimeoutRecovered = flow.recover { case t: TimeoutException ⇒ throw new HttpConnectionTimeoutException(t.getMessage) }
+        val flowWithTimeoutRecovered = flow.via(MapError { case t: TimeoutException ⇒ new HttpConnectionTimeoutException(t.getMessage) })
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala]: Ah! Thanks to @drewhk I see why that's the fix. So if we throw in a stage we log, and if we invoke failure we do not log automagically, and then the proper debug/error logging applies.

Thanks a lot @jrudolph for the fix!

**Comment by drewhk** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, thanks!

**Comment by ktoso** [comment]
We could consider adding a `mapError` operator actually, WDYT @drewhk @rkuhn ?

**Comment by danielwegener** [comment]
@jrudolph TIL! cheers!

**Comment by Antwnis** [comment]
Cheers! Looking forward for a 2.4.3 release

**Pull request merged**